src/main/java/org/apache/sling/resourceresolver/impl/CommonResourceResolverFactoryImpl.java

```diff
@@ -472,6 +473,22 @@ public boolean hasVanityPathPrecedence() {
         return configs;
     }
 
+    @Override
+    public List<String> getAliasPath() {
+        final AtomicReferenceArray<String> includes = this.activator.getOptimizedAliasResolutionAllowList();
+        if (includes == null) {
+            return Collections.emptyList();
+        }
+
+        final List<String> configs = new ArrayList<>();
```

Comment:
I recommend using a `TreeSet`. It provides sorting and eliminates duplicates. Also, I would avoid manual iteration. Instead if `this.activator.getOptimizedAliasResolutionAllowList()` returns a `String[]` you could write `new TreeSet<>(Arrays.asList(includes))`; or if it returns a `Collection` you could just copy it into a `TreeSet` using the constructor.

Reply:
@jsedding 
"this.activator.getOptimizedAliasResolutionAllowList()"  return AtomicReferenceArray, as we require atomicity while reading data from config. We can not make "this.activator.getOptimizedAliasResolutionAllowList()" return String[] Since sonar bot suggestion does not allow to have Non-primitive fields as "volatile" and throw a bug in the PR.

With AtomicReferenceArray, we can not use treeSet as it throw "java.lang.ClassCastException:" we may require some comparable method to do so.